### PR TITLE
Add multiple enhancements and fixes

### PR DESCRIPTION
**Pull request by korotkevics**
* Fix templates as requested by @makoJohn 
* Mention dropped features in README.md
* Solve UTF-8 related UI issue (Russian and Polish characters weren't displayed properly)
* Clear game board each time randomization is triggered
* Adjust flags' position (weren't displayed properly)
* Update Lombok version to the latest one
* Add a Sonar link to README.md
* Create REQUIREMENTS.md
* Replace each appearance of "destructed" with "destroyed" (in the code)
* Introduce the possibility of placing the fleet randomly more than just once (if a player does not like a placement he can repeat the action) 

_The below template was generated automatically. Please check in all the related check-boxes. You are welcome to add any extra remarks above this sentence._

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: server/src/test/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetPlacementProcessorTest.java [hunk 1]
```diff
@@ -38,16 +40,38 @@ public void shouldCreateRandomFleetWhenTheHeaderIsRandomPlacement() {
...
     Message message = new MessageBuilder()
         .withAuthor(Author.CLIENT)
         .withHeader(Header.RANDOM_PLACEMENT)
         .build();
-    when(communicationBus.receive(wrappedClient)).thenReturn(message);
-    FleetPlacementProcessor fleetPlacementProcessor = new FleetPlacementProcessor(communicationBus);
+
+    Message ready = manualPlacementMessage();
+
+    Message randomFleet = new MessageBuilder()
```
Comment (makoJohn) [inline server/src/test/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetPlacementProcessorTest.java]: randomFleetMessage

File: server/src/test/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetPlacementProcessorTest.java [hunk 2]
```diff
@@ -38,16 +40,38 @@ public void shouldCreateRandomFleetWhenTheHeaderIsRandomPlacement() {
     //given
     CommunicationBus communicationBus = mock(CommunicationBus.class);
     WrappedClient wrappedClient = mock(WrappedClient.class);
+    FleetGenerator fleetGenerator = mock(FleetGenerator.class);
+    Fleet fleet = mock(Fleet.class);
     Message message = new MessageBuilder()
```
Comment (makoJohn) [inline server/src/test/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetPlacementProcessorTest.java]: what kind of message is this?

File: server/src/test/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetPlacementProcessorTest.java [hunk 3]
```diff
@@ -38,16 +40,38 @@ public void shouldCreateRandomFleetWhenTheHeaderIsRandomPlacement() {
...
+    when(communicationBus.receive(wrappedClient)).thenReturn(message, ready);
+    when(fleetGenerator.generateFleet()).thenReturn(fleet);
+    FleetPlacementProcessor fleetPlacementProcessor = new FleetPlacementProcessor(communicationBus,
+        fleetGenerator);
     //when
-    Fleet generatedFleet = fleetPlacementProcessor.placeFleet(wrappedClient);
+    fleetPlacementProcessor.placeFleet(wrappedClient);
     //then
-    Assert.assertNotNull(generatedFleet);
+    verify(communicationBus).send(wrappedClient, randomFleet);
```
Comment (makoJohn) [inline server/src/test/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetPlacementProcessorTest.java]: do you need all arrangements to verify this condition?

File: client/src/main/java/pl/korotkevics/ships/client/gui/util/UTF8Control.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      try {
+        // Only this line is changed to make it to read properties files as UTF-8.
+        bundle = new PropertyResourceBundle(new InputStreamReader(stream, ENCODING));
+      } finally {
+        stream.close();
+      }
+    }
+    return bundle;
+  }
+}
```
Comment (makoJohn) [inline client/src/main/java/pl/korotkevics/ships/client/gui/util/UTF8Control.java]: add a new line and DITTO

File: client/src/main/java/pl/korotkevics/ships/client/gui/controllers/FleetPlacementController.java [hunk 5]
```diff
@@ -372,19 +372,56 @@ private void loadGameWindow() {
...
+    this.disableDragAndDropForGroupOfShips(this.groupTwoMastShips, disable);
+    this.disableDragAndDropForGroupOfShips(this.groupOneMastShips, disable);
+  }
+
+  private void disableDragAndDropForGroupOfShips(Group shipsGroup, boolean disable) {
+    for (Node ship : shipsGroup.getChildren()) {
+      ship.setDisable(disable);
+      for (Node child : ((Group) ship).getChildren()) {
+        Rectangle rec = (Rectangle) child;
+        if(disable)
```
Comment (makoJohn) [inline client/src/main/java/pl/korotkevics/ships/client/gui/controllers/FleetPlacementController.java]: always use brackets

File: server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetPlacementState.java [hunk 6]
```diff
@@ -61,8 +63,21 @@ private boolean verifyClientsAreConnected() {
   }
 
   private void placeFleet() {
-    fleets.add(this.fleetProcessor.placeFleet(this.turnManager.getCurrentPlayer()));
-    fleets.add(this.fleetProcessor.placeFleet(this.turnManager.getOtherPlayer()));
+    final Map<Integer, Fleet> clientsFleet = new ConcurrentHashMap<>();
```
Comment (makoJohn) [inline server/src/main/java/pl/korotkevics/ships/server/gamestates/fleetplacement/FleetPlacementState.java]: consider adding a comment about what is happening here

File: client/src/main/java/pl/korotkevics/ships/client/gui/util/UTF8Control.java (hunk 4 above)
Comment (korotkevics) [reply client/src/main/java/pl/korotkevics/ships/client/gui/util/UTF8Control.java]: thx! I'll include it in the next PR

**Pull request merged**